### PR TITLE
TC integration

### PR DESCRIPTION
**Pull request by g-k**
WIP

refs: #1 

Seeing this error, so I presumably need to do some sort of auth:

```
taskcluster.exceptions.TaskclusterRestFailure: You do not have sufficient scopes. This request requires you                                 
to have one of the following sets of scopes:                          
[                                  
  [                                
    "queue:declare-provisioner:test-tc-provisioner#actions"           
  ]                                
]                                  

You only have the scopes:          
[]                                 

In other words you are missing scopes from one of the options:        
 * Option 0:                       
    - "queue:declare-provisioner:test-tc-provisioner#actions"         
----                               
method:     declareProvisioner     
errorCode:  InsufficientScopes     
statusCode: 403                    
time:       2017-11-21T23:45:13.520Z 
```

### CODE REVIEW

**Comment by g-k** [comment]
@helfi92 Can I use `<workerId>` as a query param for a POST action? How do I request TC creds to test this out?

**Comment by helfi92** [comment]
@g-k Actions don't use query parameters anymore. The relevant params are baked inside the path instead. To give you an example, for the aws provisioner, we have an action with url: `https://ec2-manager.taskcluster.net/v1/region/<workerGroup>/instance/<workerId>`. 

You can definitely use `<workerId>` in the url. Before sending a request, we do the following replacements to an action of `context=worker`: https://github.com/taskcluster/taskcluster-tools/blob/master/src/views/WorkerManager/WorkerManager.js#L168-L172

To know whether a request is permitted, You can make a call to `auth.authenticateHawk` with the hawk authorization header and then verify that the returned scopes satisfy the scopes required. See [backend services](https://docs.taskcluster.net/manual/using/integration/backend) for more details.

To request TC credentials, see how to [acquire taskcluster credentials for running tests](https://docs.taskcluster.net/reference/workers/generic-worker#acquire-taskcluster-credentials-for-running-tests).

**Comment by g-k** [comment]
Thanks @helfi92!

I updated the API to use action URLs like `/api/v1/workers/<workerId>/group/<workerGroup>/jobs?task_name=reboot`. Will the static query param `task_name` be OK or should I move that into the path?

**Comment by helfi92** [comment]
Definitely. Feel free to have static query parameters.
